packages/nuxt/src/core/nuxt.ts

```diff
@@ -124,18 +124,28 @@ async function initNuxt (nuxt: Nuxt) {
   await nuxt.callHook('modules:before')
   const modulesToInstall = [...nuxt.options.modules]
 
+  const watchedPaths = new Set<string>()
+
   // Automatically register user modules
   for (const config of nuxt.options._layers.map(layer => layer.config)) {
```

Comment:
We probably want to reverse the other of layers. A module in base module should be ready by the time user (higher order module) is executed

Reply:
If that's the case, wouldn't we also expect that those modules have been run before explicitly specified modules in nuxt.config (e.g. unshift these modules)?

Reply:
I think that also make sense! It should be last layer > ... > first layer > user > ad-hoc

Reply:
sounds good